pkg/csi/service/vanilla/controller.go

```diff
@@ -679,19 +679,16 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 				},
 			})
 			if err != nil {
-				msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
-				log.Error(msg)
-				return nil, status.Error(codes.Internal, msg)
-			}
-			if len(queryResult.Volumes) == 0 {
-				msg := fmt.Sprintf("volumeID %s not found in QueryVolume", req.VolumeId)
+				msg := fmt.Sprintf("QueryAllVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
 				log.Error(msg)
 				return nil, status.Error(codes.Internal, msg)
 			}
-			if queryResult.Volumes[0].VolumeType == common.BlockVolumeType {
-				volumeType = prometheus.PrometheusBlockVolumeType
-			} else {
-				volumeType = prometheus.PrometheusFileVolumeType
+			if len(queryResult.Volumes) == 1 {
```

Comment:
volume type will be empty when timed out delete is called again, are we ok with Prometheus registering the delete matrics without volume type?

Reply:
volumetype is initialized as unknown. So, we are registering the metrics with the volumeType all the time. We try to find the volumeType if we can.